public/app/features/logs/logsModel.ts

```diff
@@ -92,13 +92,11 @@ export function dedupLogRows(rows: LogRowModel[], strategy?: LogsDedupStrategy):
   }
 
   return rows.reduce((result: LogRowModel[], row: LogRowModel, index) => {
-    const rowCopy = { ...row };
```

Comment:
Just a minor change to not clone every row by default. Marginal improvement.

Reply:
If something is mutating the dataframes this will introduce bugs that are hard to track down

Reply:
You can see in line 99 that we still create a new object, but only when it's needed, not for every row including duplicates, which aren't pushed to "result".